src/pudl/transform/eiaaeo.py

```diff
@@ -178,6 +241,41 @@ def _decumulate(df: pd.DataFrame, decumulate_columns: list[str]) -> pd.DataFrame
             columns=[f"cumulative_{c}" for c in decumulate_columns]
         )
 
+    def _check_totals_add_up(unstacked) -> None:
+        """Check that reported totals match with calculated totals.
+
+        We want to check that:
+
+        * the "total" value is equal to the sum of the non-total values for
+          "cumulative planned/unplanned additions/retirements."
+        * cumulative planned + unplanned additions == cumulative total additions
+        """
+        ratio_close_reported_calculated = subtotals_match_reported_totals_ratio(
+            unstacked.reset_index(),
+            pk=["report_year", "model_case_eiaaeo", "region", "projection_year"],
+            fact_columns=[
+                "cumulative_planned_additions",
+                "cumulative_unplanned_additions",
+                "cumulative_retirements",
+            ],
+            dimension_column="dimension",
+        )
+        assert (
+            0.999 < ratio_close_reported_calculated <= 1.0
+        ), f"reported vs. calculated totals: {ratio_close_reported_calculated}"
+
+        totals = unstacked.xs("total", level="dimension")
+        ratio_close_additions_to_total = series_sum_ratio(
+            summands=totals.loc[
+                :, ["cumulative_planned_additions", "cumulative_unplanned_additions"]
+            ],
+            total=totals.cumulative_total_additions,
+        )
```

Comment:
I thought about that - since we don't actually know what the future call sites look like I leaned towards keeping this thing generic. Once we start seeing "oh, this setup is a pain" we can easily pull that into `series_sum_ratio` - but IME it's harder to say "oh, actually we want different setup here vs. in the other place, now we have to refactor `series_sum_ratio`."